community/modules/compute/schedmd-slurm-gcp-v5-partition/main.tf

```diff
@@ -60,13 +60,18 @@ locals {
       instance_template = null
     },
   ]
+
+  # Since deployment name may be used to create a cluster name, we remove any invalid character from the begining
+  # Also, slurm imposed a lot of restrictions to this name, so we format it to an acceptable string
+  tmp_name           = lower(substr(replace(var.deployment_name, "/^[^a-z]*|[^a-z0-9]/", ""), 0, 8))
```

Comment:
This would remove any upper case letters before lower gets a chance to do anything if I'm reading this right. I'd run lower first, then replace then substr.

Reply:
nit: `tmp_name` isn't very descriptive. I'd go with `filtered-deployment-name` or `filtered-deploy` or something like that.

Reply:
characters (a-z) are not being touched by the regexp. It removes any non character happening in the beginning of the string, and anything that is not a character or a number happening anywhere. 
The reason I do it last is that since the substr already cuts to just the first 8 characters, lower has to run on a smaller string...

Reply:
changed tmp_name to tmp_cluster_name

Reply:
sorry, I misunderstood your comment on the regexp... you are right.. I fixed it now.